syntax = "proto3";

service XArm {
    rpc move_jspace_path (Path) returns (Status) {}
    rpc get_jnt_values (Empty) returns (JntValues) {}
    rpc jaw_to (GripperStatus) returns (Status) {}
    rpc get_gripper_status (Empty) returns (GripperStatus) {}
}

message Empty {
}

message Status {
  enum StatusValue {
    ERROR = 0;
    DONE = 1;
  }
  StatusValue value = 1;
}

message JntValues {
  bytes data = 1;
}

message Path {
    int32  length = 1;
    int32  njnts = 2;
    bytes  data = 3;
}

message GripperStatus {
    int32 speed = 1;
    int32 position = 2;
}